crates/astria-composer/src/executor/mod.rs

```diff
@@ -607,6 +567,76 @@ async fn submit_tx(
     res
 }
 
+/// Reports and logs exit reason
+#[instrument(skip_all, err)]
+fn report_exit(reason: &eyre::Result<&str>) -> eyre::Result<()> {
+    match &reason {
+        Ok(reason) => {
+            info!(reason, "shutting down");
+            Ok(())
+        }
+        Err(reason) => {
+            error!(%reason, "shutting down");
+            Err(eyre!(reason.to_string()))
+        }
+    }
+}
+
+/// Handles timeout of shutdown process
+#[instrument(skip_all)]
+async fn bundle_drain_timout_handler(shutdown_logic: impl Future<Output = eyre::Result<()>>) {
+    match tokio::time::timeout(BUNDLE_DRAINING_DURATION, shutdown_logic).await {
+        Ok(Ok(())) => info!("executor shutdown tasks completed successfully"),
+        Ok(Err(error)) => error!(%error, "executor shutdown tasks failed"),
+        Err(error) => error!(%error, "executor shutdown tasks failed to complete in time"),
```

Comment:
The tokio error here is an `Elapsed` instance, which [doesn't give any more useful info than our own string](https://github.com/tokio-rs/tokio/blob/694577fa8563dfe62878a4b3b4879a2199e3c409/tokio/src/time/error.rs#L111-L115).
```suggestion
        Err(error) => error!("executor shutdown tasks failed to complete in time"),
```